hledger/Hledger/Cli/CliOptions.hs

```diff
@@ -181,7 +181,9 @@ reportflags = [
 
   -- generated postings/transactions
  ,flagNone ["auto"]          (setboolopt "auto") "apply automated posting rules to modify transactions"
- ,flagNone ["forecast"]      (setboolopt "forecast") "generate future transactions from periodic transaction rules, for the next 6 months or till report end date. In hledger-ui, also make ordinary future transactions visible."
+ ,flagNone ["forecast"]      (setboolopt "forecast") "generate future transactions from periodic transaction rules, for the next 6 months or till report end date (could be overriden by --forecast-begin/--forecast-end). In hledger-ui, also make ordinary future transactions visible."
+ ,flagReq  ["forecast-begin"]     (\s opts -> Right $ setopt "forecast-begin" s opts) "DATE" "include forecast postings/txns on or after this date"
+ ,flagReq  ["forecast-end"]       (\s opts -> Right $ setopt "forecast-end" s opts) "DATE" "include forecast postings/txns before this date"
```

Comment:
I like that you put "forecast" first, grouping these flags.

I wanted to improve the help somehow. What do you think of:
```
--forecast        generate periodic transactions (from periodic transaction rules).
                  By default these begin after the latest recorded transaction,
                  and end 6 months from today, or at the report end date.
                  Also, in hledger-ui, make future transactions visible.

--forecast-begin  generate periodic transactions from this date

--forecast-end    generate periodic transactions up to this date
```

These seem slightly redundant and verbose right now
(hledger --forecast --forecast-begin=...), I wonder if we can simplify.

Reply:
Eg (brainstorming):
```
--forecast | --forecast-in PERIODEXPR
--forecast [forecastdate:PERIODEXPR]  
```

Reply:
Or, just say that using any of the three flags enables periodic transactions ? (hledger --forecast-begin ...  is enough).

Reply:
Yet another:
```
--forecast-period PERIODEXPR
```

Reply:
For posterity: i took the help text and tried to shoot for `--forecast[=PERIODICEXPR]`